### PR TITLE
br: clean codes -- simplify the `CreateTable` stage

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve?

Issue Number: ref #52877

Problem Summary:
we can clean some codes
### What changed and how does it work?
simplify the `CreateTable` stage
### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/task/restore.go [hunk 1]
```diff
@@ -1067,30 +1069,25 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
...
-			return t
-		})
+		for _, createdTable := range createdTables {
+			cfg.tiflashRecorder.Rewrite(createdTable.OldTable.Info.ID, createdTable.Table.ID)
+		}
 	}
 
-	// Block on creating tables before restore starts. since create table is no longer a heavy operation any more.
-	tableStream = client.GoBlockCreateTablesPipeline(ctx, maxRestoreBatchSizeLimit, tableStream)
+	tableStream := afterTableCreatedCh(ctx, createdTables)
```
Comment (Leavrth) [inline br/pkg/task/restore.go:1090]: removed by the next PR

File: br/pkg/restore/snap_client/pipeline_items.go [hunk 2]
```diff
@@ -390,112 +386,6 @@ func concurrentHandleTablesCh(
...
-		}
-		if err != nil {
-			errCh <- err
-		}
-	}()
-
-	return outCh
-}
-
-func (rc *SnapClient) GoBlockCreateTablesPipeline(ctx context.Context, sz int, inCh <-chan CreatedTable) <-chan CreatedTable {
```
Comment (Leavrth) [inline br/pkg/restore/snap_client/pipeline_items.go:473]: move this function to `br/pkg/task/restore.go: afterTableCreatedCh`

File: br/pkg/restore/snap_client/pipeline_items.go [hunk 3]
```diff
@@ -390,112 +386,6 @@ func concurrentHandleTablesCh(
 	}
 }
 
-// GoCreateTables create tables, and generate their information.
-// this function will use workers as the same number of sessionPool,
-// leave sessionPool nil to send DDLs sequential.
-func (rc *SnapClient) GoCreateTables(
```
Comment (Leavrth) [inline br/pkg/restore/snap_client/pipeline_items.go:396]: move this function to `br/pkg/restore/snap_client/client.go: CreateTables`.

Maybe we can create a new file called `create_tables.go` to save the client functions and wrap the part of steps from `br/pkg/task/restore.go: runRestore`.

File: br/pkg/task/restore.go [hunk 4]
```diff
@@ -1067,30 +1069,25 @@ func runRestore(c context.Context, g glue.Glue, cmdName string, cfg *RestoreConf
 		}
 
 		// Hijack the tableStream and rewrite the rewrite rules.
-		tableStream = util.ChanMap(tableStream, func(t snapclient.CreatedTable) snapclient.CreatedTable {
+		for _, createdTable := range createdTables {
```
Comment (Leavrth) [inline br/pkg/task/restore.go:1072]: Now `createdTable` is a pointer

File: br/pkg/restore/snap_client/client.go [hunk 5]
```diff
@@ -672,12 +672,43 @@ func (rc *SnapClient) getRebasedTables() map[restore.UniqueTableName]bool {
...
+		}
+		// fall back to old create table (sequential create table)
+		log.Info("fall back to the sequential create table")
+	}
+
+	// restore tables in db pool
+	if len(rc.dbPool) > 0 {
+		return rc.createTablesSingle(ctx, rc.dbPool, tables, newTS)
+	}
+	// restore tables in one db
```
Comment (3pointer) [inline br/pkg/restore/snap_client/client.go:702]: Can this `non-dbPool` part be removed?

Comment (Leavrth) [reply br/pkg/restore/snap_client/client.go:702]: we can unify the `rc.dbPool` and `rc.db` by another PR.

**Review by 3pointer — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**